### PR TITLE
Rf/fix/explorer urls v2

### PR DESCRIPTION
**Pull request by RafaDev95**
Removed some network and explorrer env variables

https://app.clickup.com/t/86a56p7gm

### CODE REVIEW

File: src/modules/transactions/components/TransactionCard/DetailsDialog.tsx [hunk 1]
```diff
@@ -58,7 +58,7 @@ const DetailsDialog = ({ ...props }: DetailsDialogProps) => {
   const handleViewInExplorer = () => {
     const { hash, network } = transaction;
     window.open(
-      `${findBlockExplorerByNetwork(network.url)}/tx/0x${hash}`,
+      `${findNetworkByUrl(network.url).explorer}/tx/0x${hash}`,
```
Comment (luisburigo) [inline src/modules/transactions/components/TransactionCard/DetailsDialog.tsx:61]: prevent reference error on network not inserted in storage

```ts
const handleViewInExplorer = () => {
    const { hash, network } = transaction;
    const networkInfo = findNetworkByUrl(network.url);
    
    if(networkInfo) {
      window.open(
        `${networkInfo.explorer}/tx/0x${hash}`,
        '_BLANK'
      );
    }
```

**Review by luisburigo — CHANGES_REQUESTED** [review]

File: src/modules/transactions/components/TransactionCard/deposit-details/DepositDetails.tsx [hunk 2]
```diff
@@ -23,12 +22,11 @@ const DepositDetails = ({ transaction }: DepositDetailsProps) => {
   const {
     screenSizes: { isMobile, isLowerThanFourHundredAndThirty },
   } = useWorkspaceContext();
-  const { currentNetwork } = useNetworks();
 
   const handleViewInExplorer = () => {
     const { hash, network } = transaction;
     window.open(
-      `${findBlockExplorerByNetwork(network.url)}/tx/0x${hash}`,
+      `${findNetworkByUrl(network.url).explorer}/tx/0x${hash}`,
```
Comment (luisburigo) [inline src/modules/transactions/components/TransactionCard/deposit-details/DepositDetails.tsx:29]: https://github.com/infinitybase/bako-safe-ui/pull/485#discussion_r1792609911

File: src/modules/vault/components/box/index.tsx [hunk 3]
```diff
@@ -84,7 +84,7 @@ const VaultBox = (props: VaultBoxPropx) => {
 
   const redirectToNetwork = () =>
     window.open(
-      `${findBlockExplorerByNetwork(network.url)}/account/${address}/assets`,
+      `${findNetworkByUrl(network.url).explorer}/account/${address}/assets`,
```
Comment (luisburigo) [inline src/modules/vault/components/box/index.tsx:87]: https://github.com/infinitybase/bako-safe-ui/pull/485#discussion_r1792609911

File: src/modules/transactions/providers/toast.tsx [hunk 4]
```diff
@@ -23,7 +22,7 @@ const useTransactionToast = () => {
 
   const handleViewInExplorer = (hash: string, networkUrl: string) => {
     window.open(
-      `${findBlockExplorerByNetwork(networkUrl)}/tx/0x${hash}`,
+      `${findNetworkByUrl(networkUrl).explorer}/tx/0x${hash}`,
```
Comment (luisburigo) [inline src/modules/transactions/providers/toast.tsx:25]: https://github.com/infinitybase/bako-safe-ui/pull/485#discussion_r1792609911

**Review by luisburigo — CHANGES_REQUESTED** [review]

File: src/modules/transactions/components/TransactionCard/Details.tsx [hunk 5]
```diff
@@ -46,7 +46,7 @@ const Details = ({
   const handleViewInExplorer = () => {
     const { hash, network } = transaction;
     window.open(
-      `${findBlockExplorerByNetwork(network.url)}/tx/0x${hash}`,
+      `${findNetworkByUrl(network.url).explorer}/tx/0x${hash}`,
```
Comment (luisburigo) [inline src/modules/transactions/components/TransactionCard/Details.tsx:49]: https://github.com/infinitybase/bako-safe-ui/pull/485#discussion_r1792609911

**Review by luisburigo — CHANGES_REQUESTED** [review]

File: src/modules/network/services/methods.ts [hunk 6]
```diff
@@ -46,39 +42,33 @@ export type DeleteNetworkResponse = void;
...
-  return network?.explorer ?? defaultBlockExplorer;
+};
+
+export const findNetworkByUrl = (url: string) => {
+  const network = Object.values(availableNetWorks).find(
+    (network) => network.url === url,
+  );
+
+  return network ?? availableNetWorks[NetworkType.MAINNET];
 };
```
Comment (guimroque) [inline src/modules/network/services/methods.ts:72]: we have an default return, `findByNetwork` will never return undefined

Comment (guimroque) [reply src/modules/network/services/methods.ts:72]: @luisburigo

**Review by guimroque — APPROVED** [review]

**Pull request merged**